src/aks-preview/azext_aks_preview/decorator.py

```diff
@@ -2346,6 +2455,22 @@ def set_up_azure_keyvault_kms(self, mc: ManagedCluster) -> ManagedCluster:
 
         return mc
 
+    def set_up_apiserver_vnet_integration(self, mc: ManagedCluster) -> ManagedCluster:
+        """Set up apiserverAccessProfile enableVnetIntegration and subnetId for the ManagedCluster object.
+
+        :return: the ManagedCluster object
+        """
+        if self.context.get_enable_apiserver_vnet_integration():
+            if mc.api_server_access_profile is None:
+                mc.api_server_access_profile = self.models.ManagedClusterAPIServerAccessProfile()
+            mc.api_server_access_profile.enable_vnet_integration = True
+        if self.context.get_apiserver_subnet_id():
+            if mc.api_server_access_profile is None:
+                mc.api_server_access_profile = self.models.ManagedClusterAPIServerAccessProfile()
+            mc.api_server_access_profile.subnet_id = self.context.get_apiserver_subnet_id()
+
+        return mc
```

Comment:
Maybe inherit and extend the functions named `set_up_api_server_access_profile` and `update_api_server_access_profile`? To me, this is the controller function used to set up api_server_access_profile.

Reply:
fixed